### PR TITLE
feature/11.x Reject and Filter Method for defer collection instance

### PR DESCRIPTION
**Pull request by danuiachimovschi**
**Implementation of filter and reject Methods for Defer: Enhancing Middleware Flexibility**

Benefits to End Users:

Customizable Request and Response Handling: These methods enable users to easily filter or exclude specific requests or responses based on custom conditions, making middleware implementation more powerful and intuitive.

Improved Workflow Control: By supporting conditional logic, users can streamline their application's request and response flow without resorting to verbose or repetitive code.

### CODE REVIEW

File: tests/Support/SupportCollectionTest.php [hunk 1]
```diff
@@ -5590,14 +5590,12 @@ public function testPercentageReturnsNullForEmptyCollections($collection)
...
      */
-    public static function collectionClassProvider()
+    public static function collectionClassProvider(): \Generator
     {
-        return [
-            [Collection::class],
-            [LazyCollection::class],
-        ];
+        yield [Collection::class];
+        yield [LazyCollection::class];
```
Comment (shaedrich) [inline tests/Support/SupportCollectionTest.php:5598]: Is this change related to defer? oO

Comment (danuiachimovschi) [reply tests/Support/SupportCollectionTest.php:5598]: Not really, it's just a minor optimization.

Comment (shaedrich) [reply tests/Support/SupportCollectionTest.php:5598]: Not sure if that would be better suited for a separate PR 🤔

Comment (danuiachimovschi) [reply tests/Support/SupportCollectionTest.php]: I will rollback it!

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!
